### DISCUSSION TITLE
No Option to choose our own AAD APP for permission Management for PnP Search WebParts.

### DISCUSSION THREAD

**Discussion by INNMPJ** [microsoft-search/pnp-modern-search#1807] [Ideas]
The problem we have encountered with the approval of required permissions is like whatever permissions we have granted for the PnP Search Webparts are available to other webparts (created by someone else) on the SharePoint  page which we want to avoid. 
We are looking for an option where we can provide our own ADD APP ID instead of using default AAD app which is been registered by PnP Search.
I request to enable this feature.

@wobba

**Comment by FranckyC** [comment]
Hi @INNMPJ, this is of course the default behavior for API access with SPFx. In the solution we only declare permissions that are useful to use the Web Parts but these are not mandatory. That's why you should approve only needed permissions first or approve your own permissions in the API Access portal (https://microsoft-search.github.io/pnp-modern-search/installation/).

To get the behavior you need, you can still repackage the solution using the `isDomainIsolated` flag to turn them to isolated Web Parts https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/isolated-web-parts.

We don't plan to switch to isolated Web Parts in a near future so I'll close this one for now.

**Reply by wobba** [reply]
@FranckyC moved it to discussions as an enhancement to allow adding a separate AAD ID and not use the OOB SPFx one which is shared among all SPFx web parts. It's not about iframe isolation, but about knowing what each web part uses of Graph permissions - as the OOB SPFx model is a bit flawed.
